# STEP 2: Data Collection and Documentation 

In [21]:
import pandas as pd


# load the algorithm dataset
alg_df = pd.read_csv("data/raw/KEN_2012-2023_RTC_v01_M_CSV/ma3route_crashes_algorithmcode.csv")

In [22]:
alg_df.shape

(31064, 10)

In [23]:
alg_df.head()

,crash_id,crash_datetime,crash_date,latitude,longitude,n_crash_reports,contains_fatality_words,contains_pedestrian_words,contains_matatu_words,contains_motorcycle_words
0,1,2018-06-06 20:39:54,2018-06-06,-1.263030,36.764374,1,0,0,0,0
1,2,2018-08-17 06:15:54,2018-08-17,-0.829710,37.037820,1,1,0,0,0
2,3,2018-05-25 17:51:54,2018-05-25,-1.125301,37.003297,1,0,0,0,0
3,4,2018-05-25 18:11:54,2018-05-25,-1.740958,37.129026,1,0,0,0,0
4,5,2018-05-25 21:59:54,2018-05-25,-1.259392,36.842321,1,1,0,0,0


In [24]:
alg_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 31064 entries, 0 to 31063
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   crash_id                   31064 non-null  int64  
 1   crash_datetime             31064 non-null  str    
 2   crash_date                 31064 non-null  str    
 3   latitude                   31064 non-null  float64
 4   longitude                  31064 non-null  float64
 5   n_crash_reports            31064 non-null  int64  
 6   contains_fatality_words    31064 non-null  int64  
 7   contains_pedestrian_words  31064 non-null  int64  
 8   contains_matatu_words      31064 non-null  int64  
 9   contains_motorcycle_words  31064 non-null  int64  
dtypes: float64(2), int64(6), str(2)
memory usage: 2.4 MB


In [25]:
alg_df.describe()

,crash_id,latitude,longitude,n_crash_reports,contains_fatality_words,contains_pedestrian_words,contains_matatu_words,contains_motorcycle_words
count,31064.000000,31064.000000,31064.000000,31064.000000,31064.000000,31064.000000,31064.000000,31064.000000
mean,15532.500000,-1.272481,36.852499,1.400914,0.073526,0.030389,0.081799,0.036763
std,8967.548717,0.118961,0.113650,1.486540,0.261002,0.171658,0.274062,0.188182
min,1.000000,-3.100000,36.283950,1.000000,0.000000,0.000000,0.000000,0.000000
25%,7766.750000,-1.316142,36.798093,1.000000,0.000000,0.000000,0.000000,0.000000
50%,15532.500000,-1.281033,36.840281,1.000000,0.000000,0.000000,0.000000,0.000000
75%,23298.250000,-1.244783,36.891427,1.000000,0.000000,0.000000,0.000000,0.000000
max,31064.000000,-0.565402,37.879490,66.000000,1.000000,1.000000,1.000000,1.000000


In [26]:
# Check columns for alg_df

alg_df.columns

Index(['crash_id', 'crash_datetime', 'crash_date', 'latitude', 'longitude',
       'n_crash_reports', 'contains_fatality_words',
       'contains_pedestrian_words', 'contains_matatu_words',
       'contains_motorcycle_words'],
      dtype='str')

In [27]:
# load the manual_code dataset

manual_df = pd.read_csv("./data/raw/KEN_2012-2023_RTC_v01_M_CSV/ma3route_crashes_manualcode.csv")
manual_df.head()

,crash_id,crash_datetime,crash_date,latitude,longitude,n_crash_reports,contains_fatality_words,contains_pedestrian_words,contains_matatu_words,contains_motorcycle_words
0,1,2018-06-06 20:39:54,2018-06-06,-1.263132,36.764283,1,0,0,0,0
1,2,2018-06-30 13:36:54,2018-06-30,-1.295135,36.821932,1,0,0,0,0
2,3,2018-05-26 07:42:54,2018-05-26,-1.371855,36.920457,1,1,0,0,0
3,4,2018-05-26 07:52:24,2018-05-26,-1.209945,36.833230,1,0,0,0,0
4,5,2018-05-26 15:42:24,2018-05-26,-1.215468,36.850638,1,0,0,0,0


In [28]:
manual_df.shape

(2595, 10)

In [29]:
manual_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2595 entries, 0 to 2594
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   crash_id                   2595 non-null   int64  
 1   crash_datetime             2595 non-null   str    
 2   crash_date                 2595 non-null   str    
 3   latitude                   2595 non-null   float64
 4   longitude                  2595 non-null   float64
 5   n_crash_reports            2595 non-null   int64  
 6   contains_fatality_words    2595 non-null   int64  
 7   contains_pedestrian_words  2595 non-null   int64  
 8   contains_matatu_words      2595 non-null   int64  
 9   contains_motorcycle_words  2595 non-null   int64  
dtypes: float64(2), int64(6), str(2)
memory usage: 202.9 KB


In [30]:
# check the columns for manula_df

manual_df.columns

Index(['crash_id', 'crash_datetime', 'crash_date', 'latitude', 'longitude',
       'n_crash_reports', 'contains_fatality_words',
       'contains_pedestrian_words', 'contains_matatu_words',
       'contains_motorcycle_words'],
      dtype='str')

## Dataset Description

Source: Kenya Road Traffic Crashes Dataset (2012–2023)

Variables in both datasets:

- crash_id: Unique crash identifier
- crash_datetime: Date and time of crash
- crash_date: Date of crash
- latitude: Crash latitude
- longitude: Crash longitude
- n_crash_reports: Number of reports
- contains_fatality_words: Indicates fatality-related crash
- contains_pedestrian_words: Indicates pedestrian involvement
- contains_matatu_words: Indicates matatu involvement
- contains_motorcycle_words: Indicates motorcycle involvement

### Considering the column names match, I will have to combine the datasets

In [31]:
krtc_df = pd.concat([alg_df,manual_df], ignore_index=True)

krtc_df

,crash_id,crash_datetime,crash_date,latitude,longitude,n_crash_reports,contains_fatality_words,contains_pedestrian_words,contains_matatu_words,contains_motorcycle_words
0,1,2018-06-06 20:39:54,2018-06-06,-1.263030,36.764374,1,0,0,0,0
1,2,2018-08-17 06:15:54,2018-08-17,-0.829710,37.037820,1,1,0,0,0
2,3,2018-05-25 17:51:54,2018-05-25,-1.125301,37.003297,1,0,0,0,0
3,4,2018-05-25 18:11:54,2018-05-25,-1.740958,37.129026,1,0,0,0,0
4,5,2018-05-25 21:59:54,2018-05-25,-1.259392,36.842321,1,1,0,0,0
...,...,...,...,...,...,...,...,...,...,...
33654,3111,2018-05-23 07:23:54,2018-05-23,-1.434153,36.965546,4,0,0,1,0
33655,3112,2018-05-23 14:38:24,2018-05-23,-1.329874,36.869561,3,0,0,0,0
33656,3113,2018-05-24 07:45:24,2018-05-24,-1.269935,36.792622,18,1,0,1,0
33657,3114,2018-05-24 08:12:24,2018-05-24,-1.325156,36.848661,2,0,0,0,0


In [32]:
# To verify the merge

krtc_df.shape

(33659, 10)

### After the merge, we now have 33659 rows and 10 columns

# STEP 3: Data Understanding

In [33]:
# check for null values

krtc_df.isnull().sum()

crash_id                     0
crash_datetime               0
crash_date                   0
latitude                     0
longitude                    0
n_crash_reports              0
contains_fatality_words      0
contains_pedestrian_words    0
contains_matatu_words        0
contains_motorcycle_words    0
dtype: int64

In [34]:
# check for duplicates

krtc_df.duplicated().sum()

np.int64(0)

#### There are no NAN or duplicates in the combined dataset

In [35]:
# save the combined dataset

krtc_df.to_csv("./data/processes/ma3route_crashes_combined.csv", index = False)